sdk/keyvault/keyvault-keys/review/keyvault-keys.api.md

```diff
@@ -502,6 +511,16 @@ export interface RsaEncryptParameters {
 // @public
 export type SignatureAlgorithm = string;
 
+// @public
+export interface SignDataOptions extends CryptographyOptions {
+    hashAlgorithm?: string;
```

Comment:
I think we should make this an enum like we do for signature algorithms. Having free-form input like this will be difficult for any language to support. Remember that we have to make decisions on this with `signData` and `verifyData` to know how to actually has the data.

Reply:
And just to confirm, you're saying that - currently - the service only accepts 32 bytes so only a SHA256 is supported?

Reply:
Just a quick clarification: the service will accept _up to_ 32 bytes - SHA256 or SHA224 would work. I don't remember if SHA128 does, but for sure anything that produces over 32 bytes will get rejected by the service. 

An extensible enough would make sense!

Reply:
We'll also need to consider how that works for hash algorithm-mandated signature algorithms like ES256. In those cases, do we simply ignore the optional hash algorithm (if optional) or err if one is specified? This is one thing the "parameters" pattern tried to establish: what options were suitable for which algorithms. I agree that *right now* it'd be overkill, but I'd hate to design a pattern now that won't work in the future.

/cc @tg-msft